src/De/DeJson.jl

```diff
@@ -2,15 +2,326 @@ module DeJson
 
 export deser_json
 
-using ..ParJson
+using YYJSON
+import ..Serde
+import ..Serde: deser, eldeser, isempty, custom_name, default_value, nulltype
+import ..Serde: WrongType, CustomType, NullType, PrimitiveType, ArrayType, DictType, NTupleType
 import ..to_deser
 
+# Any
+
+function deser(::Type{Any}, val_ptr::Ptr{YYJSONVal})
+    return if yyjson_is_str(val_ptr)
+        unsafe_string(yyjson_get_str(val_ptr))
+    elseif yyjson_is_raw(val_ptr)
+        unsafe_string(yyjson_get_raw(val_ptr))
+    elseif yyjson_is_real(val_ptr)
+        yyjson_get_real(val_ptr)
+    elseif yyjson_is_int(val_ptr)
+        yyjson_get_int(val_ptr)
+    elseif yyjson_is_bool(val_ptr)
+        yyjson_get_bool(val_ptr)
+    elseif yyjson_is_obj(val_ptr)
+        deser(Dict{String,Any}, val_ptr)
+    elseif yyjson_is_arr(val_ptr)
+        deser(Vector{Any}, val_ptr)
+    end
+end
+
+function deser(::Type{T}, ::Type{Union{Nothing,E}}, val_ptr::Ptr{YYJSONVal}) where {T,E}
+    return deser(T, E, val_ptr)
+end
+
+function deser(::Type{T}, ::Type{Nothing}, val_ptr::Ptr{YYJSONVal}) where {T}
+    return deser(Nothing, val_ptr)
+end
+
+function deser(::Type{T}, ::Type{E}, val_ptr::Ptr{YYJSONVal}) where {T,E}
+    return deser(E, val_ptr)
+end
+
+# PrimitiveType
+
+function deser(::PrimitiveType, ::Type{T}, val_ptr::Ptr{YYJSONVal}) where {T<:AbstractString}
+    return if yyjson_is_str(val_ptr)
+        unsafe_string(yyjson_get_str(val_ptr))
+    elseif yyjson_is_raw(val_ptr)
+        unsafe_string(yyjson_get_raw(val_ptr))
+    elseif yyjson_is_real(val_ptr)
+        repr(yyjson_get_num(val_ptr))
+    elseif yyjson_is_int(val_ptr)
+        repr(Int64(yyjson_get_num(val_ptr)))
+    elseif yyjson_is_bool(val_ptr)
+        repr(yyjson_get_bool(val_ptr))
+    end
+end
```

Comment:
```suggestion
function deser(::PrimitiveType, ::Type{T}, value_ptr::Ptr{YYJSONVal}) where {T<:AbstractString}
    return if yyjson_is_str(value_ptr)
        unsafe_string(yyjson_get_str(value_ptr))
    elseif yyjson_is_raw(value_ptr)
        unsafe_string(yyjson_get_raw(value_ptr))
    else
        throw(WrongType("Expected a string for type `AbstractString`."))
    end
end
```

Reply:
WrongType expects a lot of arguments so decided to throw an julia `error`.